File: backend/app/models/transcript.py
"""
Pydantic v2 models for the transcript API endpoints.

Provides response shapes for the three transcript endpoints:
  - GET /api/transcript/{debate_id}/speeches        -> TranscriptResponse
  - GET /api/transcript/{debate_id}/speech/{id}     -> SpeechTextResponse
  - GET /api/transcript/{debate_id}/suggestions     -> SuggestionsResponse
"""
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class TranscriptSpeechRow(BaseModel):
    """One speech occurrence in chronological order for transcript viewer."""

    speech_id: str
    phase_id: str
    phase_title: str
    speaker_id: str
    first_name: str
    last_name: str
    party: Optional[str] = None
    speaking_role: Optional[str] = None
    is_government_member: bool = False


class TranscriptResponse(BaseModel):
    """All speeches for a debate, chronologically ordered."""

    debate_id: str
    debate_title: str
    session_date: str          # ISO date string for breadcrumb
    session_id: str            # For breadcrumb link
    chamber: str               # "camera" or "senato"
    speeches: list[TranscriptSpeechRow]


class SpeechTextResponse(BaseModel):
    """Text content of a single speech (lazy-loaded on accordion expand)."""

    speech_id: str
    text: str


class SuggestionsResponse(BaseModel):
    """Starter questions for the debate chatbot."""

    questions: list[str]

File: frontend/src/types/transcript.ts
/**
 * TypeScript interfaces for the Debate Transcript Viewer feature.
 * These mirror the backend Pydantic models in backend/app/models/transcript.py
 */

export interface TranscriptSpeechRow {
  speech_id: string;
  phase_id: string;
  phase_title: string;
  speaker_id: string;
  first_name: string;
  last_name: string;
  party: string | null;
  speaking_role: string | null;
  is_government_member: boolean;
}

export interface TranscriptResponse {
  debate_id: string;
  debate_title: string;
  session_date: string;
  session_id: string;
  chamber: string;
  speeches: TranscriptSpeechRow[];
}

export interface SpeechTextResponse {
  speech_id: string;
  text: string;
}

export interface SuggestionsResponse {
  questions: string[];
}

export interface SearchMatch {
  speech_id: string;
  snippet: string;
}

export interface TranscriptSearchResponse {
  query: string;
  matches: SearchMatch[];
}

/** Chat message for the transcript chatbot (session-only, no persistence). */
export interface TranscriptMessage {
  id: string;
  role: "user" | "assistant";
  content: string;
  citations?: TranscriptCitation[];
}

/** Citation from the debate chatbot pointing to a specific speech. */
export interface TranscriptCitation {
  index: number;         // [1], [2], etc.
  speech_id: string;
  speaker_name: string;
  party: string | null;
  chunk_text: string;    // relevant excerpt
}
